src/arch/riscv64/Emit.zig

```diff
@@ -63,6 +63,9 @@ pub fn emitMir(emit: *Emit) Error!void {
 
                         hi_r_type = Elf.R_ZIG_GOT_HI20;
                         lo_r_type = Elf.R_ZIG_GOT_LO12;
+                    } else if (sym.flags.needs_got) {
+                        hi_r_type = Elf.R_GOT_HI20_STATIC; // TODO this needs reworking here and for x86_64
```

Comment:
"TODO this is needs to be rewritten" is a conclusion with a missing argument. Code comments should stick to providing facts and explanations, and avoid vague wishful thinking. Discussion about whether something should be rewritten or not belongs on the issue tracker, not in TODO comments.

#363

Reply:
Fair point. Will open an issue and link to it.

Reply:
That's mainly a TODO for me and added by me. Re-engineering a solution at this stage is difficult as requires redesigning the concept of `.got.zig`.

Reply:
That's fine to delay re-engineering a solution. However, the comment should either explain the problem, link to an open issue, or be deleted.